src/extract.rs

```diff
@@ -67,8 +67,21 @@ pub struct Extract<R, C> {
     columns: C,
 }
 
-impl<R, C> Extract<R, C> {
-    pub fn new(rows: R, columns: C) -> Extract<R, C> {
+impl<R, C> Extract<R, C>
+where
+    R: RangeBounds<usize>,
+    C: RangeBounds<usize>,
+{
+    pub fn rows(rows: R) -> Extract<R, RangeFull> {
+        Extract {
+            rows: rows,
+            columns: ..,
+        }
+    }
+    pub fn columns(columns: C) -> Extract<RangeFull, C> {
+        Extract { rows: .., columns }
+    }
+    pub fn segment(rows: R, columns: C) -> Extract<R, C> {
         Extract { rows, columns }
     }
```

Comment:
What do you think will it be better to separate the factions like this?

```rust
impl<R, C> Extract<R, C>
where
    R: RangeBounds<usize>,
    C: RangeBounds<usize>,
{
    pub fn segment(rows: R, columns: C) -> Self {
        Extract { rows, columns }
    }
}

impl<R> Extract<R, RangeFull>
where
    R: RangeBounds<usize>,
{
    pub fn rows(rows: R) -> Self {
        Extract {
            rows: rows,
            columns: ..,
        }
    }
}

impl<C> Extract<RangeFull, C>
where
    C: RangeBounds<usize>,
{
    pub fn columns(columns: C) -> Self {
        Extract { rows: .., columns }
    }
}
```

Reply:
I made this change as suggested. 

I'm new and would love to know what advantages this approach offers 😃